go/arrow/array/data.go

```diff
@@ -190,9 +190,33 @@ func (d *Data) SetDictionary(dict arrow.ArrayData) {
 	}
 }
 
+// SizeInBytes returns the size of the Data and any children and/or dictionary in bytes by recursively examining the nested structures of children and/or dictionary.
+// The value returned is an upper-bound since offset is not taken` into account.`
```

Comment:
Just to keep consistency in trying width of the comments for easier reading:

```suggestion
// SizeInBytes returns the size of the Data and any children and/or dictionary in bytes by 
// recursively examining the nested structures of children and/or dictionary.
// The value returned is an upper-bound since offset is not taken into account.
```